### PR TITLE
[Keyboard] Dumbpad v3x

### PR DESCRIPTION
**Pull request by deveth0**
## Description

This PR adds support for the new v3.x of dumbpad by @imchipwood  (https://github.com/imchipwood/dumbpad) which features per-key RGB and hotswap sockets.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/dumbpad/v3x/templates/keymap.c
Comment (zvecr) [inline keyboards/dumbpad/v3x/templates/keymap.c:1]: This should be removed, and `encoder_update_kb` implemented in `v3x.c`instead

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/dumbpad/v3x/templates/keymap.c
Comment (deveth0) [reply keyboards/dumbpad/v3x/templates/keymap.c]: Thanks for you fast feedback, done :)

File: keyboards/dumbpad/v3x/v3x.c
Comment (drashna) [inline keyboards/dumbpad/v3x/v3x.c:97]:
```suggestion
#ifdef ENCODER_ENABLE
bool encoder_update_user(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
```

File: keyboards/dumbpad/v3x/v3x.c
Comment (drashna) [inline keyboards/dumbpad/v3x/v3x.c:119]:
```suggestion
}
#endif
```

File: keyboards/dumbpad/v3x/keymaps/default/keymap.c
Comment (drashna) [inline keyboards/dumbpad/v3x/keymaps/default/keymap.c:83]:
```suggestion
```

File: keyboards/dumbpad/v3x/keymaps/deveth0/keymap.c
Comment (drashna) [inline keyboards/dumbpad/v3x/keymaps/deveth0/keymap.c:106]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/dumbpad/v3x/v3x.c
Comment (deveth0) [reply keyboards/dumbpad/v3x/v3x.c:97]: @drashna  / @zvecr sorry, I mixed up the `encoder_update_user` and `encoder_update_kb` implementations, should be ok now :)

* `encoder_update_kb` => in keyboard.c, call `encoder_update_user`
* `encoder_update_user` => in keymaps

correct (similar to planck).

File: keyboards/dumbpad/v3x/keymaps/default/keymap.c
Comment (deveth0) [reply keyboards/dumbpad/v3x/keymaps/default/keymap.c]: as this is quite specific implementation for the second layer, I opted to only implement the volume in keyboard.c and add the additional functionallity to the keymap, hope that's ok

**Comment by deveth0** [comment]
Hey, I think all open issues should be resolved, could you please have another look and let me know, how to proceed?
Cheers

**Review by drashna — APPROVED** [review]

**Comment by deveth0** [comment]
I updated the PR to include VIA Mappings

**Comment by deveth0** [comment]
Any chance to get a second review / merge?

File: keyboards/dumbpad/v3x/v3x.c
Comment (fauxpark) [inline keyboards/dumbpad/v3x/v3x.c:101]:
```suggestion
bool led_update_kb(led_t led_state) {
    if (!led_update_user(led_state)) return;
    // put your keyboard LED indicator (ex: Caps Lock LED) toggling code here
    writePin(LED_02, !led_state.num_lock);
}
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/dumbpad/v3x/v3x.c
Comment (deveth0) [reply keyboards/dumbpad/v3x/v3x.c]: Thanks, good finding on the newer API :) Fixed

**Comment by deveth0** [comment]
Bumb :)

**Comment by imchipwood** [comment]
Bump - I believe all requested changes have been implemented.

**Comment by deveth0** [comment]
Bump ?

File: keyboards/dumbpad/v3x/readme.md
Comment (noroadsleft) [inline keyboards/dumbpad/v3x/readme.md:27]:
```suggestion
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the encoder and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
Everything here looks good except for the sequence of the readme, but I don't see why this PR is targeting the `develop` branch.

**Comment by dickofn** [comment]
why is this PR closed and not merge while https://github.com/qmk/qmk_firmware/pull/17058/checks is merged

**Comment by noroadsleft** [comment]
The PR author never responded to my question or committed the changes I requested.

**Comment by dickofn** [comment]
> The PR author never responded to my question regarding why the PR targets the `develop` branch, nor committed the changes I requested.

@noroadsleft may I create a new fork and pr with the changes you suggest? I'm new to this kind of open-source contribution

**Comment by noroadsleft** [comment]
> may I create a new fork and pr with the changes you suggest? I'm new to this kind of open-source contribution

Re-implementing this PR is okay – especially so if you have the supported hardware, or some kind of relationship with the original PR's author (though not strictly required).

**Comment by dickofn** [comment]
> > may I create a new fork and pr with the changes you suggest? I'm new to this kind of open-source contribution
> 
> Re-implementing this PR is okay – especially so if you have the supported hardware, or some kind of relationship with the original PR's author (though not strictly required).

I already did re-implementing all https://github.com/qmk/qmk_firmware/pull/18663

to be honest no connection to anyone, but have the product and very happy using it since the first time

**Pull request closed without merging**
